agent/agent.go

```diff
@@ -414,11 +416,10 @@ func (a *agent) reportMetadataLoop(ctx context.Context) {
 	}
 }
 
-// reportLifecycleLoop reports the current lifecycle state once.
-// Only the latest state is reported, intermediate states may be
-// lost if the agent can't communicate with the API.
+// reportLifecycleLoop reports the current lifecycle state once. All state
+// changes are reported in order.
```

Comment:
Review: Originally, this was a weird/performance choice to only submit the latest status which simply resulted in needing to document the behavior. We now change it so that the agent always reports all states, we achieve this by adding the timestamp for the event in the payload.